File: mobile/src/repositories/onboardingRepository.ts
import {
  markHouseholdCreatedGuideSeen,
  markJoinWelcomeSeen,
  markShoppingListGuideSeen,
  resetHouseholdOnboarding,
  shouldShowHouseholdCreatedGuide,
  shouldShowJoinWelcome,
  shouldShowShoppingListGuide,
} from '../local/onboardingStorage';

export const onboardingRepository = {
  shouldShowHouseholdCreatedGuide,
  markHouseholdCreatedGuideSeen,
  shouldShowJoinWelcome,
  markJoinWelcomeSeen,
  shouldShowShoppingListGuide,
  markShoppingListGuideSeen,
  resetHouseholdOnboarding,
};

File: app/src/main/java/com/salino/sali/domain/repository/OnboardingRepository.kt
package com.salino.sali.domain.repository

interface OnboardingRepository {
    suspend fun shouldShowHouseholdCreatedGuide(): Boolean
    suspend fun markHouseholdCreatedGuideSeen()
    suspend fun shouldShowJoinWelcome(): Boolean
    suspend fun markJoinWelcomeSeen()
    suspend fun shouldShowShoppingListGuide(): Boolean
    suspend fun markShoppingListGuideSeen()
    suspend fun resetHouseholdOnboarding()
}
